go/apps/mixnet/mixnet_router/mixnet_router.go

```diff
@@ -68,16 +69,28 @@ var x509Identity pkix.Name = pkix.Name{
 
 func main() {
 	flag.Parse()
-	timeout, _ := time.ParseDuration("5s") // TODO(cjpatton) make this a command line parameter.
-	hp, err := mixnet.NewRouterContext(*configPath, *serverNetwork, *serverAddr, *batchSize,
+	timeout, err := time.ParseDuration(*timeoutDuration)
+	if err != nil {
+		glog.Errorf("router: failed to parse timeout duration: %s", err)
+	}
+
+	hp, err := mixnet.NewRouterContext(*configPath, *routerNetwork, *routerAddr, *batchSize,
 		timeout, &x509Identity, tao.Parent())
 	if err != nil {
-		glog.Errorf("failed to configure server: %s", err)
+		glog.Errorf("failed to configure router: %s", err)
 	}
-	defer hp.Close()
 
-	if err = serveMixnetClients(hp); err != nil && err != io.EOF {
-		glog.Errorf("error occured while serving: %s", err)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGKILL)
+	go func() {
+		sig := <-sigs
+		hp.Close()
+		glog.Infof("router: closing on signal: %s", sig)
+		os.Exit(0)
```

Comment:
POSIX requires that processes that are terminated by a signal return with an exit code that is greater than 128. Conventionally, this is done by adding the signal number to 128. However, os.Signal is an interface, so there's no way to get the signal number other than registering different signal handlers for each signal. So, I don't mind if you just return 0x81 for now.